nml/spritecache.py

```diff
@@ -137,17 +133,21 @@ def read_cache(self):
         """
         Read the *.grf.cache[index] files.
         """
-        if not (os.access(self.cache_filename, os.R_OK) and os.access(self.cache_index_filename, os.R_OK)):
-            # Cache files don't exist
+
+        try:
+            with generic.open_cache_file(self.sources, ".cache", 'rb') as cache_file:
+                index_file = generic.open_cache_file(self.sources, ".cacheindex", 'r')
+
+                cache_data = array.array('B')
+                cache_size = os.fstat(cache_file.fileno()).st_size
+                cache_data.fromfile(cache_file, cache_size)
+                assert cache_size == len(cache_data)
+                self.cache_time = os.path.getmtime(cache_file.name)
+        except OSError:
+            # Cache files don't exist (or otherwise aren't readable)
             return
 
-        index_file = open(self.cache_index_filename, 'r')
-        cache_file = open(self.cache_filename, 'rb')
-        cache_data = array.array('B')
-        cache_size = os.fstat(cache_file.fileno()).st_size
-        cache_data.fromfile(cache_file, cache_size)
-        assert cache_size == len(cache_data)
-        self.cache_time = os.path.getmtime(self.cache_filename)
+        # cache_file is closed by `with` block, but index_file is still open.
```

Comment:
This is less than ideal, the close() is way too far down. But at least this change makes it better.